src/apps/app_webhook.erl

```diff
@@ -0,0 +1,126 @@
+-include("ar_config.hrl").
+
+-define(NUMBER_OF_TRIES, 10).
+-define(WAIT_BETWEEN_TRIES, 30 * 1000).
+
+-define(BASE_HEADERS, [
+	{<<"content-type">>, <<"application/json">>}
+]).
+
+start(Configs) ->
+	adt_simple:start(?MODULE, Configs).
+
+new_block(Configs, Block) ->
+	call_webhooks(Block, Configs),
+	Configs.
+
+new_transaction(Configs, Tx) ->
+	call_webhooks(Tx, Configs),
+	Configs.
+
+call_webhooks(Entity, Configs) ->
+	lists:foreach(
+		fun(#config_webhook { events = Events } = Config) ->
+			ok = case lists:member(entity_type(Entity), Events) of
+				true -> try_call_webhook(Entity, Config);
+				false -> ok
+			end
+		end,
+		Configs
+	).
+
+try_call_webhook(Entity, Config) ->
+	try_call_webhook(Entity, Config, 0).
+
+try_call_webhook(Entity, Config, N) when N < ?NUMBER_OF_TRIES ->
+	#config_webhook{
+		url = Url,
+		headers = Headers,
+		include_data = IncludeData
+	} = Config,
+	try
+		{ok, {Scheme, _UserInfo, Host, Port, Path, Query}} = http_uri:parse(Url),
+		{ok, Client} = fusco:start(atom_to_list(Scheme) ++ "://" ++ binary_to_list(Host) ++ ":" ++ integer_to_list(Port), []),
+		{ok, Result} =
+			fusco:request(
+				Client,
+				<<Path/binary, Query/binary>>,
+				<<"POST">>,
+				?BASE_HEADERS ++ Headers,
+				to_json(Entity, IncludeData),
+				10000
+			),
+		case Result of
+			{{<<"200">>, _}, _, _, _, _} ->
+				ok = fusco:disconnect(Client),
+				ar:info([successful_webhook_call, {
+					{type, entity_type(Entity)},
+					{id, entity_id(Entity)},
+					{url, Url},
+					{headers, Headers},
+					{response, Result}
+				}]),
+				Url;
+			UnsuccessfulResult ->
+				ar:warn([unsuccessful_webhook_call,
+					{type, entity_type(Entity)},
+					{id, entity_id(Entity)},
+					{url, Url},
+					{headers, Headers},
+					{response, UnsuccessfulResult},
+					{retry_in, ?WAIT_BETWEEN_TRIES}
+				]),
+				timer:sleep(?WAIT_BETWEEN_TRIES),
+				try_call_webhook(Entity, Config, N+1)
+		end,
+		ok
+	catch
+		Class:Exception ->
+			ar:warn([unexpected_webhook_call_error,
+				{type, entity_type(Entity)},
+				{id, entity_id(Entity)},
+				{url, Url},
+				{headers, Headers},
+				{class, Class},
+				{exception, Exception},
+				{retry_in, ?WAIT_BETWEEN_TRIES}
+			]),
+			timer:sleep(?WAIT_BETWEEN_TRIES),
+			try_call_webhook(Entity, Config, N+1)
+	end;
+try_call_webhook(Entity, Config, _) ->
+	#config_webhook{
+		url = Url,
+		headers = Headers
+	} = Config,
+	ar:warn([gave_up_webhook_call,
+		{type, entity_type(Entity)},
+		{id, entity_id(Entity)},
+		{url, Url},
+		{headers, Headers},
+		{number_of_tries, ?NUMBER_OF_TRIES},
+		{wait_between_tries, ?WAIT_BETWEEN_TRIES}
+	]),
+	ok.
+
+entity_type(#block {}) -> block;
+entity_type(#tx {}) -> transaction.
+
+entity_id(#block { indep_hash = Id }) -> ar_util:encode(Id);
+entity_id(#tx { id = Id }) -> ar_util:encode(Id).
+
+to_json(#block {} = Block, _) ->
+	{JsonKVPairs} = ar_serialize:block_to_json_struct(Block),
+	JsonStruct = {lists:keydelete(wallet_list, 1, JsonKVPairs)},
```

Comment:
`wallet_list`s in the serialized blocks are hashes now, not sure if it's worth deleting them.

Reply:
Ah, no, a hash is computed when it's stored, here we do get the full list so nevermind.